### DISCUSSION TITLE
Un système de Doorlock ?

### DISCUSSION THREAD

**Discussion by Rotabla42** [SOZ-Faut-etre-Sub/SOZ-FiveM-Server#330] [Q&A] [answered]
J'ai peut-être mal cherché, mais je n'ai pas trouvé de système de verrouillage de portes. 
La possibilité à créer des serrures pouvant être ouvertes ou fermer par un job, des joueurs spécifiques ou des gangs. 

Je me demandais si c'était une fonctionnalité qui n'était pas dans l'OSS ou simplement que vous n'en avez pas forcément l'utilité
(Même si on peut entrer dans les mappings comme dans un moulin, les joueurs de Soz se restreignent eux-mêmes).

Ça n'a rien à voir avec le titre de la question, mais je me demandais aussi, vu qu'il y a des entreprises qui ne sont pas à jour dans le wiki, s'il était possible de proposer une page pour les entreprises manquantes (UPW, PAWL, Demetal Comp) ou si c'était réservé à la communauté de Soz ?

Personnellement, pour trouver les informations, j'ai parcouru les fichiers liés aux entreprises pour trouver ce que je cherchais et parfois déduit le fonctionnement de certaines fonctionnalités.
 Je me dis que ça pourrait faire gagner du temps à ceux qui n'ont pas joué sur le serveur d'avoir un guide complet.

Merci d'avance !

**Comment by Laikker** [comment]
Hello,
Pour le Doorlock, c'est une fonctionnalité qui n'est pas actuellement dans l'OSS

**Reply by Rotabla42** [reply]
D'accord merci de ta réponse !
Au niveau du wiki c'est possible ou il faut voir ça via un ticket sur discord ?

**Comment by Darkwolf-dev** [comment]
Bonjour,

Avec la nouvelle mise à jour de l'OSS, est ce que le système d'installation de verrous sur les portes et des droits par métier présent sur Soz est également présent ou bien il n'est pas dans l'oss ?
j'ai vu dans le menu développeur la possibilité de cocher l'option : 
![image](https://github.com/user-attachments/assets/788e44d2-5d78-48ca-9a19-98944ef69c52)

Mais ensuite, en tentant de B-target des portes, rien n'est possible.

Ou alors, je m'y prends mal ?

Merci

**Reply by Laikker** [reply]
![image](https://github.com/user-attachments/assets/096781c1-46ba-4d7e-940b-81d270a6a2c2)
Si tu as l'option activé et le rôle admin ou staff, tu as ce barget sur les portes.

**Reply by Darkwolf-dev** [reply]
J'ai bien l'option activé de mon côté et je pense que c est good pour le rôle admin mais de mon côté le B-target ne donne rien : 
![image](https://github.com/user-attachments/assets/b00548be-8357-4b31-aca0-0e2295fb25cf)

**Reply by Darkwolf-dev** [reply, marked as answer]
En commentant ce petit bout de code, cela fonctionne bien
![image](https://github.com/user-attachments/assets/fb57644b-ff7a-46f7-bb87-0019980d8144)

Du fait de l'absence de la bdd externe pour les compte soz_api, les grades de comptes ne sont pas valorisé pour le getPlayer()
Tout fonctionne bien sinon !

Merci beaucoup pour l'aide !
